plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+		baseAsset      hProtocol.Asset
+		quoteAsset     hProtocol.Asset
+		db             *sql.DB
+		config         *VolumeFilterConfig
+
+		wantError        error
+		wantSubmitFilter SubmitFilter
+	}{
+		// Note: config can be additional market IDs, make sure to test additionalMarketIds and optionalAccountIDs
+		{
+			name:             "failure - display base",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "FAIL", Quote: ""},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        hProtocol.Asset{},
+			quoteAsset:       hProtocol.Asset{},
+			db:               db,
+			config:           &VolumeFilterConfig{},
+			wantError:        fmt.Errorf("could not convert base asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:             "failure - display quote",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "XLM", Quote: "FAIL"},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        testNativeAsset,
+			quoteAsset:       hProtocol.Asset{},
+			config:           &VolumeFilterConfig{},
+			wantError:        fmt.Errorf("could not convert quote asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:             "failure - query",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        testNativeAsset,
+			quoteAsset:       testNativeAsset,
+			config:           &VolumeFilterConfig{},
+			db:               nil,
+			wantError:        fmt.Errorf("could not make daily volume by date Query: could not make daily volume by date action: the provided db should be non-nil"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:           "success - empty config",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         testEmptyVolumeConfig,
+			db:             db,
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 testEmptyVolumeConfig,
+				dailyVolumeByDateQuery: emptyConfigVolumeQuery,
+			},
+		},
+		{
+			name:           "success - market ids",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         &VolumeFilterConfig{additionalMarketIDs: marketIDs},
```

Comment:
most of the configs used in these test cases are invalid -- see comment above about valid configs.